### PR TITLE
Twitter handler

### PR DESCRIPTION
**Pull request by adrika-novrialdi**
Add functionality for sending images from twitter's link. This implementation is using twitter API for python. 

In general, the implementation add more configuration needed for environment variable (the env needed are listed in README.md). 

I also move some functionality that used by both of the handler into a new file. 

This implementation also has supported NSFW filter from twitter but it is not 100% accurate. At this implementation, if that tweet is categorized as NSFW, all the image in the tweet will be spoiler-ed.

### CODE REVIEW

File: botofculture/twitter_service.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+TWITTER_TOKEN_KEY = config.get('TWITTER_TOKEN_KEY')
+TWITTER_TOKEN_SECRET = config.get('TWITTER_TOKEN_SECRET')
+
+TWITTER_URL_PATTERN = re.compile(r'twitter.com\/.*status\/(\d*)')
+
+api = twitter.Api(consumer_key=TWITTER_CONSUMER_KEY, consumer_secret=TWITTER_CONSUMER_SECRET,
+                  access_token_key=TWITTER_TOKEN_KEY, access_token_secret=TWITTER_TOKEN_SECRET)
+
+
+async def handle(message: discord.message):
```
Comment (addianto) [inline botofculture/twitter_service.py]: The type hint in the parameter should be `message: discord.Message`.

File: botofculture/twitter_service.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            await parse_image(file_url, message, is_nsfw(status_data))
+
+
+async def get_status_data(status_id):
+    tweet_detail = api.GetStatus(status_id=status_id).AsDict()
+    if tweet_detail['media']:
+        return tweet_detail
+
+
+async def parse_image(file_url, message: discord.message, is_nsfw):
```
Comment (addianto) [inline botofculture/twitter_service.py]: The type hint for `message` parameter should be `discord.Message`. Similarly, `is_nsfw` parameter should have type hint `bool`.

File: botofculture/twitter_service.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    async with message.channel.typing():
+        await send_image(message, file_path, is_nsfw)
+
+
+def download_image(url: str):
+    file_name = wget.download(url, config.get('DOWNLOAD_PATH'))
+    return Path(os.path.abspath(file_name))
+
+
+def is_nsfw(status_data):
```
Comment (addianto) [inline botofculture/twitter_service.py]: `status_data` is a `dict` I presume?

**Review by addianto — CHANGES_REQUESTED** [review]
Please make sure all functions are properly type hinted.

**Comment by adrika-novrialdi** [comment]
Changes implemented. All Functions listed has used type hint in the parameter.

**Review by addianto — APPROVED** [review]
LGTM. 👍

**Pull request merged**